build/docker-compose.yml

```diff
@@ -16,6 +16,7 @@ services:
     tmpfs: /var/lib/mysql
     volumes:
      - "./docker/mysql.conf.d:/etc/mysql/conf.d"
+    container_name: mattermost-mysql
```

Comment:
this will fail if we have multiple build in the same worker node

Reply:
Do you use an .env file in cases where you would be doing this?

Reply:
We currently rely on `COMPOSE_PROJECT_NAME`, exported as a randomly generated value in the Jenkins build, to ensure all concurrently running builds are namespaced. Unfortunately, setting `container_name` ignores that value altogether, leading to @cpanato's concern.

One option would be to fork the docker-compose.yml file, and have one for building and one for development. It's not too onerous to ensure updates are made in both places, and it might even be nicer to have the developer one at the root.

Reply:
@lieut-data As per https://docs.docker.com/compose/extends/ we could have the base `docker-compose.yml`, a `docker-compose.override.yml` that is automatically used in development, and then a `docker-compose.prod.yml` (or some such) that is used for prod.

Reply:
Did a (very) quick update to the PR with what I was talking about it. By default it uses the override file and will create its own names if run with:
```bash
docker-compose --no-ansi -f ./build/docker-compose.yml -f ./build/docker-compose.prod.yml run --rm start_dependencies
```